app/Http/Controllers/BattalionController.php

```diff
@@ -57,16 +57,16 @@ public function show($alias)
 
         $battlead = DB::select('SELECT k.rname FROM battalion b
                                 INNER JOIN knight k ON k.pkey = b.battlead
-                                WHERE b.battalias = ?', [$alias])[0] ?? null;
+                                WHERE b.battalias = ?' AND k.pkey in(select pkey from knight where activeflg = 1 AND delflg = 0), [$alias])[0] ?? null;
```

Comment:
The query is passed to `DB::select` as a [string](https://www.php.net/manual/en/language.types.string.php) so all of it needs to be within the quotes to work.

A text editor with supported syntax highlighting, like [Notepad++](https://notepad-plus-plus.org/) or [VSCodium](https://vscodium.com/) will show you whenever you mistakenly end a string too early.